zed_components/src/zed_camera/src/zed_camera_component.cpp

```diff
@@ -3996,23 +3996,23 @@ bool ZedCamera::publishVideoDepth(rclcpp::Time& out_pub_ts)
 
   static sl::Timestamp lastZedTs = 0;  // Used to calculate stable publish frequency
 
-  size_t rgbSubnumber = 0;
-  size_t rgbRawSubnumber = 0;
-  size_t rgbGraySubnumber = 0;
-  size_t rgbGrayRawSubnumber = 0;
-  size_t leftSubnumber = 0;
-  size_t leftRawSubnumber = 0;
-  size_t leftGraySubnumber = 0;
-  size_t leftGrayRawSubnumber = 0;
-  size_t rightSubnumber = 0;
-  size_t rightRawSubnumber = 0;
-  size_t rightGraySubnumber = 0;
-  size_t rightGrayRawSubnumber = 0;
-  size_t stereoSubnumber = 0;
-  size_t stereoRawSubnumber = 0;
-  size_t depthSubnumber = 0;
-  size_t confMapSubnumber = 0;
-  size_t disparitySubnumber = 0;
+  static size_t rgbSubnumber = 0;
+  static size_t rgbRawSubnumber = 0;
+  static size_t rgbGraySubnumber = 0;
+  static size_t rgbGrayRawSubnumber = 0;
+  static size_t leftSubnumber = 0;
+  static size_t leftRawSubnumber = 0;
+  static size_t leftGraySubnumber = 0;
+  static size_t leftGrayRawSubnumber = 0;
+  static size_t rightSubnumber = 0;
+  static size_t rightRawSubnumber = 0;
+  static size_t rightGraySubnumber = 0;
+  static size_t rightGrayRawSubnumber = 0;
+  static size_t stereoSubnumber = 0;
+  static size_t stereoRawSubnumber = 0;
+  static size_t depthSubnumber = 0;
+  static size_t confMapSubnumber = 0;
+  static size_t disparitySubnumber = 0;
```

Comment:
@roncapat why did you add the "static" keyword for all those variables?

Reply:
They do not need to be zero-initialized at each loop. It is arguable whether they should have static storage or not, in this way they aren't allocated/de-allocated each time on the stack (but since they are POD types this is not a real advantage).
I can revert and just remove zero-initialization which may avoid a bunch of assembly lines from the executable.

Reply:
No, `static` is OK in this case